src/magnet_uri.cpp

```diff
@@ -75,10 +87,22 @@ namespace libtorrent {
 
 	std::string make_magnet_uri(torrent_info const& info)
 	{
-		std::string ret;
-		sha1_hash const& ih = info.info_hash();
-		ret += "magnet:?xt=urn:btih:";
-		ret += aux::to_hex(ih);
+		std::string ret = "magnet:?";
+
+		if (info.info_hash().has_v1())
+		{
+			sha1_hash const& ih = info.info_hash().v1;
+			ret += "xt=urn:btih:";
+			ret += aux::to_hex(ih);
+		}
+
+		if (info.info_hash().has_v2())
+		{
+			if (info.info_hash().has_v1()) ret += '&';
+			sha256_hash const& ih = info.info_hash().v2;
+			ret += "xt=urn:btmh:1220";
+			ret += aux::to_hex(ih);
+		}
```

Comment:
there should be a test for `make_magnet_uri()` that creates a v2-only as well as a hybrid magnet URI